src/main/webapp/scripts/perspective.js

```diff
@@ -4,14 +4,19 @@
  *  If not valid, updates DOM with an error message
  */
 
+ const NO_COMMENT_ERROR =  new Error('No comment inputted');
+ const COMMENT_LENGTH_EXCEEDED_ERROR = new Error('Comment is too long. We are currently only able to analyze comments of up to approx. 3000 characters.');
+
  function toxicityToPercent(toxicity) {
-   return (toxicity*100).toFixed(2);
+   return (toxicity*100).toFixed(2).toString() + '%';
  }
 
 function getToxicity() {
-  const comment = new Comment(document.getElementById("comment-text").value);
-  if(!comment.isValid) {
-    updateDom(comment.getError(), 'toxicity-container');
+  let comment;
+  try {
+    comment = new Comment(document.getElementById('comment-text').value);
```

Comment:
I actually don't know if this is good practice or not. To me it seems a little bit odd that you are sharing data between JS files through the DOM (at least that's what it looks you will be doing once you have the comments API ready). 

I think we want to adapt a design model for organizing your code (ex is Model View Control model). Basically you want one or more script files that gather data via APIs and another main script file that imports the API script files. The main script file's role is to get data from the other script files and manipulate the DOM. With this you would only have one script tag in the HTML for the main script file.

For your case, I think a Comment object would contain all the data the UI needs to display (for now it seems like its just the comment text and the toxicity). This means you would need to call the comments API to get the comments, then call the Perspective API to get the toxicity, then build the Comment object, and then pass the object to method that knows how to use the properties in that object to display it in the UI.

Does this makes? I feel like I'm even confusing myself.

https://stackoverflow.com/questions/950087/how-do-i-include-a-javascript-file-in-another-javascript-file
https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller

Reply:
These functions are for the Perspective API so we are not sharing data using the DOM (getting youtube comments is completely in the backend). The 'comment-text' ID is for user-inputted new comments which are the only type of comment that will be handled in the front end code. I could change the class name to something like userComment, but the two uses of comments are completely separated by frontend/backend so is it still necessary?

I refactored such that:
-Model: perspective-api-service.js - accepts requests for api call, responds with api data (toxicity)
-Model: comment-class - accepts requests to make comment object, responds with comment object that is ready for API call
-View: script.js - functions that manipulate the DOM
-Controller: perspective.js - handles loading relevant scripts, contains function that the user input button calls, sends data to user interface functions

From your description, I realize that scripts should be loaded in the View, but I don't think that makes sense for our project since we want to use the same UI functions in many html pages, but not the same js scripts. Therefore, I put script loading in the Controller instead (for now).
Also, I realize the View should be doing more work (handling an object and deciding how to display it) but since there is so little data to display right now (1 line of text) I think it would make sense to implement that later on when there is more data.

I hope this has made it easier to read, it was very fun to implement the MVC model! I imagine it'll be very helpful when Zoe adds the submit comment to Youtube feature